client/pagination.go

```diff
@@ -0,0 +1,60 @@
+package client
```

Comment:
This uses golang generics.
Please reach out if comments are missing or something is unclear.

With generics, we can easily expand it in the future to the rest of the API calls.

Reply:
correction. I had to comment out generics due to harness using go 1.17.